File: backend/app/schemas/query_history.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field

from app.schemas.common import MessageResponse, PaginationMeta


class QueryHistoryRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: int | None = None
    query_params: dict
    result_count: int
    created_at: datetime


class QueryHistoryListData(BaseModel):
    items: list[QueryHistoryRead]
    pagination: PaginationMeta


class QueryHistoryListResponse(BaseModel):
    success: bool = True
    data: QueryHistoryListData


class SaveQueryHistoryRequest(BaseModel):
    query_params: dict
    result_count: int = Field(default=0, ge=0)


class SaveQueryHistoryResponse(MessageResponse):
    history_id: int

File: frontend/src/types/queryHistory.ts
import type { ListData, MessageResponse } from './common'
import type { ProjectSearchRequest } from './project'

export interface QueryHistoryRead {
  id: number
  user_id?: number | null
  query_params: Partial<ProjectSearchRequest> & Record<string, unknown>
  result_count: number
  created_at: string
}

export interface QueryHistoryListResponse {
  success: boolean
  data: ListData<QueryHistoryRead>
}

export interface SaveQueryHistoryRequest {
  query_params: Record<string, unknown>
  result_count: number
}

export interface SaveQueryHistoryResponse extends MessageResponse {
  history_id: number
}
